File: parse_data.py
import json
import re

def parse_txt():
    # File uses UTF-8 or maybe EUC-KR, let's try UTF-8
    filepath = "c:\\Users\\HP\\.gemini\\antigravity\\영어공부앱\\# 영어 핵심 동사 100 완벽 가이드 🇺🇸.txt"
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            lines = f.readlines()
    except UnicodeDecodeError:
        with open(filepath, 'r', encoding='cp949') as f:
            lines = f.readlines()

    data = []
    
    current_chapter = ""
    current_verb_intro = ""
    current_day = None
    
    i = 0
    while i < len(lines):
        line = lines[i].strip()
        if not line:
            i += 1
            continue
            
        # Match Chapter
        if line.startswith("## CHAPTER"):
            current_chapter = line.replace("##", "").strip()
            
        # Match Verb intro: ### **HAVE — 기본 개념:**
        elif line.startswith("### **") and "기본 개념" in line:
            verb_match = re.search(r'\*\*(.*?)\s*—\s*기본 개념:\*\*', line)
            if verb_match:
                verb_name = verb_match.group(1).strip()
            else:
                verb_name = line.replace("### **", "").split("—")[0].strip()
            
            # Read explanation
            i += 1
            intro_text = []
            while i < len(lines) and not lines[i].strip().startswith("---") and not lines[i].strip().startswith("**DAY"):
                if lines[i].strip():
                    intro_text.append(lines[i].strip())
                i += 1
            current_verb_intro = " ".join(intro_text)
            continue
            
        # Match Day: **DAY 1 — HAVE ① : "소유하다 / 가지고 있다" (possession)**
        elif line.startswith("**DAY"):
            day_match = re.search(r'\*\*DAY\s+(\d+)\s+—\s+(.*?)\s*:\s*"(.*?)"\s*(?:\((.*?)\))?\*\*', line)
            if day_match:
                day_num = int(day_match.group(1))
                verb_phrase = day_match.group(2).strip()
                korean_meaning = day_match.group(3).strip()
                english_meaning = day_match.group(4).strip() if day_match.group(4) else ""
                
                # Next is usually the quote
                i += 1
                quote = ""
                if i < len(lines) and lines[i].strip().startswith(">"):
                    quote = lines[i].strip().replace(">", "").strip().replace("*", "")
                    i += 1
                
                # Next is explanation
                explanation = []
                while i < len(lines) and not lines[i].strip().startswith("- ") and not lines[i].strip().startswith("---"):
                    if lines[i].strip():
                        explanation.append(lines[i].strip().replace("*", ""))
                    i += 1
                
                # Next are bullet points (examples)
                examples = []
                while i < len(lines) and lines[i].strip().startswith("- "):
                    examples.append(lines[i].strip().replace("- ", ""))
                    i += 1
                    
                data.append({
                    "day": day_num,
                    "chapter": current_chapter,
                    "verb": verb_phrase.split(" ")[0] if " " in verb_phrase else verb_phrase, # strip numerals
                    "verb_full": verb_phrase,
                    "korean_meaning": korean_meaning,
                    "english_meaning": english_meaning,
                    "quote": quote,
                    "explanation": " ".join(explanation),
                    "examples": examples
                })
                continue
        i += 1
        
    js_content = f"const vocabData = {json.dumps(data, ensure_ascii=False, indent=2)};\n"
    with open("c:\\Users\\HP\\.gemini\\antigravity\\영어공부앱\\data.js", 'w', encoding='utf-8') as f:
        f.write(js_content)
        
    print(f"Successfully parsed {len(data)} items to data.js")

if __name__ == "__main__":
    parse_txt()

File: parse_data.js
const fs = require('fs');
const path = require('path');

function parseTxt() {
    const filePath = path.join(__dirname, '# 영어 핵심 동사 100 완벽 가이드 🇺🇸.txt');
    const content = fs.readFileSync(filePath, 'utf8');
    const lines = content.split('\n');
    
    const data = [];
    let currentChapter = "";
    let currentVerbIntro = "";
    
    let i = 0;
    while (i < lines.length) {
        const line = lines[i].trim();
        if (!line) {
            i++;
            continue;
        }
        
        if (line.startsWith('## CHAPTER')) {
            currentChapter = line.replace('##', '').trim();
        } else if (line.startsWith('### **') && line.includes('기본 개념')) {
            const verbMatch = line.match(/\*\*(.*?)\s*—\s*기본 개념:\*\*/);
            let verbName = "";
            if (verbMatch) {
                verbName = verbMatch[1].trim();
            } else {
                verbName = line.replace('### **', '').split('—')[0].trim();
            }
            
            i++;
            const introText = [];
            while (i < lines.length && !lines[i].trim().startsWith('---') && !lines[i].trim().startsWith('**DAY')) {
                if (lines[i].trim()) introText.push(lines[i].trim());
                i++;
            }
            currentVerbIntro = introText.join(' ');
            continue;
            
        } else if (line.startsWith('**DAY')) {
            const dayMatch = line.match(/\*\*DAY\s+(\d+)\s+—\s+(.*?)\s*:\s*"(.*?)"\s*(?:\((.*?)\))?\*\*/);
            if (dayMatch) {
                const dayNum = parseInt(dayMatch[1], 10);
                const verbPhrase = dayMatch[2].trim();
                const koreanMeaning = dayMatch[3].trim();
                const englishMeaning = dayMatch[4] ? dayMatch[4].trim() : "";
                
                i++;
                // Skip empty lines after DAY
                while (i < lines.length && !lines[i].trim()) {
                    i++;
                }
                
                let quote = "";
                if (i < lines.length && lines[i].trim().startsWith('>')) {
                    quote = lines[i].trim().replace('>', '').replace(/\*/g, '').trim();
                    i++;
                }
                
                const explanation = [];
                while (i < lines.length && !lines[i].trim().startsWith('- ') && !lines[i].trim().startsWith('---')) {
                    if (lines[i].trim()) explanation.push(lines[i].trim().replace(/\*/g, ''));
                    i++;
                }
                
                const examples = [];
                while (i < lines.length && lines[i].trim().startsWith('- ')) {
                    examples.push(lines[i].trim().replace('- ', ''));
                    i++;
                }
                
                data.push({
                    day: dayNum,
                    chapter: currentChapter,
                    verb: verbPhrase.includes(' ') ? verbPhrase.split(' ')[0] : verbPhrase,
                    verb_full: verbPhrase.replace(/①|②|③|④|⑤|⑥|⑦|⑧|⑨|⑩|⑪|⑫|⑬/g, '').trim(),
                    korean_meaning: koreanMeaning,
                    english_meaning: englishMeaning,
                    quote: quote,
                    explanation: explanation.join(' '),
                    examples: examples
                });
                continue;
            }
        }
        i++;
    }
    
    const jsContent = `const vocabData = ${JSON.stringify(data, null, 2)};\n`;
    fs.writeFileSync(path.join(__dirname, 'data.js'), jsContent, 'utf8');
    console.log(`Successfully parsed ${data.length} items to data.js`);
}

parseTxt();
